.github/workflows/ci.yaml

```diff
@@ -99,19 +99,6 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
```

Comment:
I know this isn't directly relevant to this PR, but actions/checkout@v4 exists now so we should upgrade to it.